dotcom-rendering/src/model/groupCards.ts

```diff
@@ -0,0 +1,90 @@
+import type {
+	DCRContainerPalette,
+	DCRContainerType,
+	DCRGroupedTrails,
+	FEFrontCard,
+} from '../types/front';
+import { enhanceCards } from './enhanceCards';
+
+/**
+ * Groups cards based on their group specified in fronts tool
+ *
+ * For 'dynamic' container types in fronts tool, cards can be grouped by sizes:
+ *  - Snap (dynamic/package only)
+ *  - Huge
+ *  - Very big
+ *  - Big
+ *  - Standard
+ *
+ * This can be consumed by the rendering layer to tell us how we should size each card, which
+ * can vary by container type.
+ *
+ * Backfilled cards are always considered 'standard'.
+ *
+ * @param curated
+ * @param backfill
+ * @param containerPalette
+ * @returns Card arrays grouped into 'snap', 'huge', 'veryBig', 'big' and 'standard'
+ */
+export const groupCards = (
+	container: DCRContainerType,
+	curated: FEFrontCard[],
+	backfill: FEFrontCard[],
+	containerPalette?: DCRContainerPalette,
+): DCRGroupedTrails => {
+	switch (container) {
+		case 'dynamic/fast':
+		case 'dynamic/slow':
+		case 'dynamic/slow-mpu':
+			return {
+				// Snap is not supported on these container types
+				snap: [],
+				huge: enhanceCards(
+					curated.filter((card) => card.card.group === '3'),
+					containerPalette,
+				),
+				veryBig: enhanceCards(
+					curated.filter((card) => card.card.group === '2'),
+					containerPalette,
+				),
+				big: enhanceCards(
+					curated.filter((card) => card.card.group === '1'),
+					containerPalette,
+				),
+				standard: enhanceCards(
+					// Backfilled cards will always be treated as 'standard' cards
+					curated
+						.filter((card) => card.card.group === '0')
+						.concat(backfill),
+					containerPalette,
+				),
+			};
+		case 'dynamic/package':
+			return {
+				huge: [],
+				veryBig: [],
+				big: [],
+				// Only 'snap' and 'standard' are supported by dynamic/package
```

Comment:
🤔 oh, wait, so dynamic package _does_ supporting some form of grouping? Interesting

Reply:
It does! I've not seen it used in a while (if at all?)

Here's what it looks like:
<img width="1323" alt="image" src="https://user-images.githubusercontent.com/9575458/181006272-5223ecdb-009b-4156-a456-40a58a2bbd14.png">
(with this set in fronts tool)
<img width="1236" alt="image" src="https://user-images.githubusercontent.com/9575458/181006333-621488a9-c5a5-484c-ab99-d2c069b7a115.png">

Here's how it renders in DCR right now:
<img width="1325" alt="image" src="https://user-images.githubusercontent.com/9575458/181006455-40d7005c-e352-4952-b657-fa2ecbf4b303.png">

This should be a simple fix for `dynamic/package` - if we update it to consume `groupedTrails` - then keep the existing logic pulling from `groupedTrails.standard` and simply add the top slice with `groupedTrails.snap`. (I've not tested what happens when you add a second snap yet - feel free to mess around with my front to test: https://m.code.dev-theguardian.com/olly)

Reply:
Thanks Olly, makes sense.

I'm guessing that this view is probably not used but if we can get DCR to support everything then in many ways this makes life easier